gcp/api/server.py

```diff
@@ -228,6 +282,23 @@ def Watch(self, request, context: grpc.ServicerContext):
     context.abort(grpc.StatusCode.UNIMPLEMENTED, "Unimplemented")
 
 
+def query_info(query) -> tuple[str, str, str]:
+  """Returns information about a query, for logging purposes.
+  First return value is one of 'commit', 'purl', 'ecosystem', 'invalid'.
+  If 'ecosystem', second two return values are the ecosystem name,
+  then 'versioned' or 'versionless' depending if the 'version' field is set.
+  Otherwise, last two return values are None.
+  """
+  if query.WhichOneof('param') == 'commit':
+    return 'commit', None, None
+  if not query.HasField('package'):
+    return 'invalid', None, None
+  if query.package.purl:
+    return 'purl', None, None
```

Comment:
Can we parse the purl here and turn it into an ecosystem query as well, since it's handled the same way dadtastore query-wise.

Reply:
+1 !

Reply:
purls seem to be handled differently?
https://github.com/google/osv.dev/blob/dcb6a15f10cc452b4fbde415e2cb1c1932878049/gcp/api/server.py#L882-L895

Reply:
Ah you're right, I got confused with the second filtering part parsing the git ecosystem. Let's keep it as is then.